aws/resource_aws_ssm_patch_baseline.go

```diff
@@ -206,35 +257,45 @@ func resourceAwsSsmPatchBaselineUpdate(d *schema.ResourceData, meta interface{})
 		params.GlobalFilters = expandAwsSsmPatchFilterGroup(d)
 	}
 
-	_, err := ssmconn.UpdatePatchBaseline(params)
-	if err != nil {
-		if isAWSErr(err, ssm.ErrCodeDoesNotExistException, "") {
-			log.Printf("[WARN] Patch Baseline %s not found, removing from state", d.Id())
-			d.SetId("")
-			return nil
+	if d.HasChange("approved_patches_enable_non_security") {
+		params.ApprovedPatchesEnableNonSecurity = aws.Bool(d.Get("approved_patches_enable_non_security").(bool))
+	}
+
+	if d.HasChange("rejected_patches_action") {
+		params.RejectedPatchesAction = aws.String(d.Get("rejected_patches_action").(string))
+	}
+
+	if d.HasChangesExcept("tags") {
+		_, err := conn.UpdatePatchBaseline(params)
+		if err != nil {
+			if isAWSErr(err, ssm.ErrCodeDoesNotExistException, "") {
+				log.Printf("[WARN] Patch Baseline %s not found, removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+			return err
```

Comment:
I know this code was just shifted down but removing a resource during an update operation will likely cause a confusing Terraform CLI error. Best to just return that error as-is instead.

```suggestion
			return fmt.Errorf("error updating SSM Patch Baseline (%s): %w", d.Id(), err)
```